### PR TITLE
Script: Write Field API path manipulation

### PR DESCRIPTION
**Pull request by stu-elastic**
Adds `WriteField` path manipulation APIs:
 * `WriteField` `move(String)`:  Moves this path to the destination path.  Throws an error if destination path exists.
 * `WriteField` `overwrite(String)`: Same as `move(String)` but overwrites the destination path if it exists;
 * `void` `remove()`: Remove the leaf value from this path.

See also: #89738
Refs: #79155

### CODE REVIEW

**Comment by stu-elastic** [comment]
@elasticsearchmachine run elasticsearch-ci/part-1

**Comment by stu-elastic** [comment]
The part-1 failure is https://github.com/elastic/elasticsearch/issues/89340

File: server/src/main/java/org/elasticsearch/script/field/WriteField.java [hunk 1]
```diff
@@ -58,30 +58,43 @@ public boolean exists() {
...
-     * @throws UnsupportedOperationException
      */
-    public void overwrite(String path) {
-        throw new UnsupportedOperationException("unimplemented");
+    public WriteField overwrite(String path) {
+        Object value = get(MISSING);
+        remove();
+        setPath(path);
+        if (value == MISSING) {
+            // remove existing mapping if it exists
```
Comment (grcevski) [inline server/src/main/java/org/elasticsearch/script/field/WriteField.java:78]: nit: should this comment say the opposite?

Comment (grcevski) [reply server/src/main/java/org/elasticsearch/script/field/WriteField.java:78]: Should we document that the overwrite will remove the path if it's empty?

**Review by grcevski — APPROVED** [review]
LGTM! I had two minor comments about documentation.

File: server/src/main/java/org/elasticsearch/script/field/WriteField.java (hunk 1 above)
Comment (stu-elastic) [reply server/src/main/java/org/elasticsearch/script/field/WriteField.java]: Fixed wording on comment.  Added javadoc about removing the path.

**Comment by stu-elastic** [comment]
@elasticsearchmachine run elasticsearch-ci/part-2

**Review by jdconrad — APPROVED** [review]
LGTM

**Pull request merged**